src/client/automation/playback/scroll.js

```diff
@@ -40,30 +37,37 @@ export default class ScrollAutomation {
         this.element         = element;
         this.offsetX         = offsetOptions.offsetX;
         this.offsetY         = offsetOptions.offsetY;
+        this.forceCenter     = offsetOptions.forceCenter;
         this.maxScrollMargin = DEFAULT_MAX_SCROLL_MARGIN;
     }
 
-    static _setScroll (element, { left, top }) {
+    _setScroll (element, { left, top }) {
+        var scrollElement       = domUtils.isHtmlElement(element) ? domUtils.findDocument(element) : element;
+        var originalElementRect = this.element.getBoundingClientRect();
+        var originalTop         = originalElementRect.top;
+        var originalLeft        = originalElementRect.left;
+        var originalScrollLeft  = styleUtils.getScrollLeft(scrollElement);
+        var originalScrollTop   = styleUtils.getScrollTop(scrollElement);
+
         left = Math.max(left, 0);
         top  = Math.max(top, 0);
 
-        if (domUtils.isHtmlElement(element)) {
-            var doc = domUtils.findDocument(element);
+        styleUtils.setScrollLeft(scrollElement, left);
+        styleUtils.setScrollTop(scrollElement, top);
 
-            styleUtils.setScrollLeft(doc, left);
-            styleUtils.setScrollTop(doc, top);
-        }
-        else {
-            styleUtils.setScrollLeft(element, left);
-            styleUtils.setScrollTop(element, top);
-        }
+        if (this.element === element || styleUtils.getScrollLeft(scrollElement) === originalScrollLeft || styleUtils.getScrollTop(scrollElement) === originalScrollTop)
+            return Promise.resolve();
+
+        return whilst(() => !this._isLayoutUpdated(originalLeft, originalTop), () => delay(50));
```

Comment:
It seems it's a way to have a forever hanging test. 
For example, 
 - we make scroll iteration
 - on the scroll event the page script moves an element to the initial client position (getBoundingClientRect stays the same)
 - we wait for 50m again and again..

Or I missed something?